app/controllers/digital-wallet/post-disable-google-pay-controller.js

```diff
@@ -0,0 +1,22 @@
+'use strict'
+
+// Local dependencies
+const paths = require('../../paths')
+const { ConnectorClient } = require('../../services/clients/connector_client')
+const auth = require('../../services/auth_service')
+const { CORRELATION_HEADER } = require('../../utils/correlation_header.js')
+const connector = new ConnectorClient(process.env.CONNECTOR_URL)
+
+module.exports = (req, res) => {
+  const gatewayAccountId = auth.getCurrentGatewayAccountId(req)
```

Comment:
Is this value not assigned to the `req` object by virtue of the `get_gateway_account` middleware? I haven't looked into this in depth just wondering why there are two potential places this could be parsed.

Reply:
@sfount I believe this is derived from the `req` object, this function just wraps the logic to fetch it: https://github.com/alphagov/pay-selfservice/blob/ceb9e51484483776a0918757e294f639f29eff74/app/services/auth_service.js#L134-L157

Reply:
Yeah I think you're correct. My comment was just that this controller sits behind middleware that (I think?) ensures there's an `account` object passed through req, so potentially you could: 

* remove the dependancy / required `auth` import 
* use the account that is passed through the middleware

```suggestion
  const gatewayAccountId = req.account.id
```